both/collections/competition.js

```diff
@@ -37,7 +30,17 @@ Competitions.attachSchema({
       afFieldInput: {
         type: "timeOfDayMinutes"
       }
-    }
+    },
+    custom: function() {
+      var obj = validationObject(this, ['calendarStartMinutes']);
+      var error = null;
+      ScheduleEvents.find({competitionId: this.docId}).forEach(function(event) {
```

Comment:
One nitty cleanup idea: `validationObject` could unconditionally copy `this.docId` into the `id` attribute of the object it creates.